docs/operations/security-overview.md

```diff
@@ -23,66 +23,109 @@ title: "Security overview"
+Then configure users, roles, and permissions, as described in the following sections. 
 
+The configuration settings mentioned below are primarily located in the `common.runtime.properties` file. Note that you need to make the configuration changes on all Druid server in the cluster.
 
-## Best practices
+## Enable TLS
 
-* Do not expose the Druid Console without authentication on untrusted networks. Access to the console effectively confers access the file system on the installation machine, via file browsers in the UI. You should use an API gateway that restricts who can connect from untrusted networks, allow list the specific APIs that your users need to access, and implements account lockout and throttling features.
-* You should only grant `WRITE` permissions to a `DATASOURCE` to trusted users. Druid assumes that these users have the same privileges as the operating system user that runs the Druid process. 
-* Grant users the minimum permissions necessary to perform their functions. For instance, do not allow users who only need to query data to write to data sources or view state.  
-* Disable JavaScript, as noted in the [Security section](https://druid.apache.org/docs/latest/development/javascript.html#security) of the JavaScript guide.
-* Run Druid as an unprivileged Unix user on the installation machine (not root).
-   > This is an important point! Administrator users on Druid have the same permission as the Unix user account it is running under. If the Druid process is running under the root user account in the OS, then Administrator users on Druid can read/write all files that the root account has access to, including sensitive files such as `/etc/passwd`.
+Enabling TLS encrypts the traffic between external clients and the Druid cluster and traffic between services within the cluster.
 
-You can configure authentication and authorization to control access to the the Druid APIs. The first step is enabling TLS for the cluster nodes. Then configure users, roles, and permissions, as described in the following sections. 
+### Generating keys
+Before you enable TLS in Druid, generate the keystore and truststore. When one Druid process, e.g. Broker, contacts another Druid process , e.g. Historical, the first service is a client for the second service, considered the server.
 
-The configuration settings mentioned below are primarily located in the `common.runtime.properties` file. Note that you need to make the configuration changes on each Druid server in the cluster. 
+The client uses a trustStore that contains certificates trusted by the client. For example, the Broker.
 
+The server uses a keyStore that contains private keys and certificate chain used to securely identify itself.
 
-## Enable TLS
+The following example demonstrates how to use Java keytool to generate the keyStore for the server and then create a trustStore to trust the key for the client:
```

Comment:
nit: Since keytool is a tool, perhaps it should be escape quotes. Maybe we should capitalize KeyStore since that's how I see it show up when I search google?

```suggestion
The following example demonstrates how to use Java `keytool` to generate the KeyStore for the server and then create a trustStore to trust the key for the client:
```